File: 26-Remove-Duplicates-from-Sorted-Array.java
import java.util.*;
class Solution {
    public int removeDuplicates(int[] nums) {
        int i = 0;
        int j = 0;
        int count =0;
        Set<Integer> numsSet = new HashSet<>();
        
        while(i<nums.length){
            if(!numsSet.contains(nums[i])){
                numsSet.add(nums[i]);
                nums[j] = nums[i];
                j ++;
                count ++;
            }
            i++;
        }
        return count;
    }
}

File: 26-Remove-Duplicates-from-Sorted-Array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i, j = 0, 1
        N = len(nums)
        if N <2:
            return N
        while j < N :
            while j < N and nums[i] == nums[j]:
                j += 1
            if j < N:
                nums[i+1] = nums[j]
                i += 1
                j += 1
        
        return i+1